packages/block-editor/src/components/link-control/index.js

```diff
@@ -43,16 +43,18 @@ const MODE_EDIT = 'edit';
 
 function LinkControl( {
 	className,
-	value,
+	value = {},
 	settings,
 	fetchSearchSuggestions,
 	instanceId,
 	onClose = noop,
 	onChange = noop,
+	popoverProps,
 } ) {
 	// State
 	const [ inputValue, setInputValue ] = useState( '' );
 	const [ isEditingLink, setIsEditingLink ] = useState( ! value || ! value.url );
```

Comment:
`! value` would now never happen as a result of the change in component signature to provide a default empty object value.

```suggestion
	const [ isEditingLink, setIsEditingLink ] = useState( ! value.url );
```

But what was the motivation behind making the change to provide a default value? Was it for convenience of direct property access on `value`?

Reply:
`value` is documented as being required, so I wonder why we need existence checks at all.

https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/link-control/README.md#value

Reply:
> But what was the motivation behind making the change to provide a default value? Was it for convenience of direct property access on value?

I think the motivation was failing tests after the refactor. It was a temporary change, I think we can remove the default value and fix the unit tests by providing a value everywhere.

Reply:
The reason this check exists in this line is different though. Basically we're saying if you're adding a new value, (empty url), the control should be in "edit mode" on mount.

Reply:
> The reason this check exists in this line is different though. Basically we're saying if you're adding a new value, (empty url), the control should be in "edit mode" on mount.

Oh, that's fine, I'm not concerned with the state or how it's initialized. I'm more concerned with the specific types we're anticipating for `value`.

Reply:
> > But what was the motivation behind making the change to provide a default value? Was it for convenience of direct property access on value?
> 
> I think the motivation was failing tests after the refactor. It was a temporary change, I think we can remove the default value and fix the unit tests by providing a value everywhere.

Initially I was inclined to agree with this, and it was my first thought as well. But as I started to implement it, I have some concern about its impact on usability.

Does it make the component less usable if I (as a consumer) am forced to construct this value object? What _is_ an initial value? Is an empty object sufficient? How am I supposed to know that?

Contrast this with the experience of any other input, which it's typically fairly straight-forward to initialize it as an undefined state and feed the `onChange` directly back into the state setter.

```jsx
const [ value, setValue ] = useState();
return <SomeInputControl value={ value } onChange={ setValue } />
```

So having a default might improve usability. I'm not sure yet if the internal implementation of the component should apply that default in the props signature, or if it might be useful to consider it before accessing any properties (to force awareness of that value shape, or allow distinct behavior depending on whether it's unset).